sorts/binary_search.cpp

```diff
@@ -0,0 +1,51 @@
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+const int32_t binarySearch(const int32_t value, const std::vector<int32_t> vector, const size_t size) {
+    if (size < 0) {
+        throw std::invalid_argument("Given argument [size] is invalid - cannot be less than minimal treshold");
+    }
+
+    uint32_t midPoint = size / 2;
+    std::cout << "Starting at position [" << midPoint << "]\n";
+    
+    while (vector[midPoint] != value) {
+        const int32_t accessedValue = vector[midPoint];
+        std::cout << "Accessed value [" << accessedValue << "] at index [" << midPoint << "]\n";
```

Comment:
Why stdout in the middle of a function? Seems like a job for a debug logger

Reply:
So, any ideas?
I've never used logging before so I have no idea how this stuff works.
Maybe something like `std::fstream logger` inside a function or something like that?

Reply:
Look up boost logging implementation